src/Orleans/CodeGeneration/TypeUtils.cs

```diff
@@ -1149,10 +1139,21 @@ private static bool DoReferencesContain(IReadOnlyCollection<AssemblyName> refere
 
         private static string EscapeIdentifier(string identifier)
         {
+            if (IsCSharpKeyword(identifier)) return "@" + identifier;
+            return identifier;
+        }
+
+        internal static bool IsCSharpKeyword(string identifier)
```

Comment:
This could be moved to another PR if desired. This method is used when escaping C# identifiers. Current we have @ symbols in front of every single identifier in generated code, this PR tries to add them only when needed. It's not perfect: some keywords are only keywords in certain contexts but we don't account for that and just always escape them.

Reply:
Roslyn has code for this: https://github.com/dotnet/roslyn/blob/master/src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/BoundNodeClassWriter.cs#L1600 maybe it's worth just to take and maintain that code as-is?